android/app/src/main/java/app/organicmaps/routing/NavigationController.java

```diff
@@ -236,4 +245,12 @@ public void onStopClicked()
     RoutingController.get().cancel();
   }
 
+  private void updateSpeedLimit(@NonNull final RoutingInfo info)
+  {
+    final Location location = LocationHelper.from(mFrame.getContext()).getSavedLocation();
+    if (location == null)
+      return;
```

Comment:
* We have location
* We show the speed limit
* We lost location
* Old speed limit will be displayed

Let's set the speed limit to zero and then return (when sl <= 0, the widget will be hidden)

Reply:
done :)